tools/scenario-player/scenario_player/runner.py

```diff
@@ -63,84 +64,41 @@ def __init__(
+        self.run_number = self.determine_run_number()
 
-        node_mode = NodeMode.EXTERNAL.name
-        if self.is_v2:
-            node_mode = nodes.get('mode', '').upper()
-            if not node_mode:
-                raise ScenarioError('Version 2 scenarios require a "mode" in the "nodes" section.')
-        try:
-            self.node_mode = NodeMode[node_mode]
-        except KeyError:
-            known_modes = ', '.join(mode.name.lower() for mode in NodeMode)
-            raise ScenarioError(
-                f'Unknown node mode "{node_mode}". Expected one of {known_modes}',
-            ) from None
+        self.node_mode = self.scenario.nodes.mode
 
         if self.is_managed:
             self.node_controller = NodeController(
                 self,
-                nodes.get('raiden_version', 'LATEST'),
-                nodes['count'],
-                nodes.get('default_options', {}),
-                nodes.get('node_options', {}),
+                self.scenario.nodes.raiden_version,
+                self.scenario.nodes.count,
+                self.scenario.nodes.default_options,
+                self.scenario.nodes.node_options,
             )
         else:
-            if 'range' in nodes:
-                range_config = nodes['range']
-                template = range_config['template']
-                self.raiden_nodes = [
-                    template.format(i)
-                    for i in range(range_config['first'], range_config['last'] + 1)
-                ]
-            else:
-                self.raiden_nodes = nodes['list']
-            self.node_commands = nodes.get('commands', {})
-
-        settings = self.scenario.get('settings')
-        if settings is None:
-            settings = {}
-        self.timeout = settings.get('timeout', TIMEOUT)
-        if self.is_managed:
-            self.protocol = 'http'
-            if 'protocol' in settings:
-                log.warning('The "protocol" setting is not supported in "managed" node mode.')
-        else:
-            self.protocol = settings.get('protocol', 'http')
-        self.notification_email = settings.get('notify')
-        self.chain_name = settings.get('chain', 'any')
+            self.raiden_nodes = self.scenario.nodes
+            self.node_commands = self.scenario.nodes.commands
 
-        if self.chain_name == 'any':
-            self.chain_name = random.choice(list(chain_urls.keys()))
-        elif self.chain_name not in chain_urls:
-            raise ScenarioError(
-                f'The scenario requested chain "{self.chain_name}" for which no RPC-URL is known.',
-            )
-        log.info('Using chain', chain=self.chain_name)
-        self.eth_rpc_urls = chain_urls[self.chain_name]
+        self.timeout = self.scenario.timeout
+        self.protocol = self.scenario.protocol
+
+        self.notification_email = self.scenario.notification_email
+
+        self.chain_name, chain_url = self.select_chain(chain_urls)
```

Comment:
IIUC `select_chain` returns `Tuple[str, List[str]]`. Therefore `chain_url` needs to be plural.